### PR TITLE
Improve thread renaming

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

File: Utilities/Thread.cpp [hunk 1]
```diff
@@ -1774,13 +1774,13 @@ void thread_base::initialize(bool(*wait_cb)(const void*))
 #endif
 
 #if defined(__APPLE__)
-	pthread_setname_np(m_name.get().substr(0, 15).c_str());
+	pthread_setname_np( m_name_cache.get()->substr(0, 15).c_str());
```
Comment (elad335) [inline Utilities/Thread.cpp]: Wild space appeared.

File: Utilities/Thread.cpp [hunk 2]
```diff
@@ -1774,13 +1774,13 @@ void thread_base::initialize(bool(*wait_cb)(const void*))
...
 #elif defined(__DragonFly__) || defined(__FreeBSD__) || defined(__OpenBSD__)
-	pthread_set_name_np(pthread_self(), m_name.get().c_str());
+	pthread_set_name_np(pthread_self(),  m_name_cache.get()->c_str());
```
Comment (elad335) [inline Utilities/Thread.cpp]: same.

File: Utilities/Thread.cpp [hunk 3]
```diff
@@ -1774,13 +1774,13 @@ void thread_base::initialize(bool(*wait_cb)(const void*))
...
 #elif defined(__NetBSD__)
-	pthread_setname_np(pthread_self(), "%s", const_cast<char*>(m_name.get().c_str()));
+	pthread_setname_np(pthread_self(), "%s", const_cast<char*>( m_name_cache.get()->c_str()));
```
Comment (elad335) [inline Utilities/Thread.cpp]: same.

File: Utilities/Thread.cpp [hunk 4]
```diff
@@ -1774,13 +1774,13 @@ void thread_base::initialize(bool(*wait_cb)(const void*))
...
 #elif !defined(_WIN32)
-	pthread_setname_np(pthread_self(), m_name.get().substr(0, 15).c_str());
+	pthread_setname_np(pthread_self(),  m_name_cache.get()->substr(0, 15).c_str());
```
Comment (elad335) [inline Utilities/Thread.cpp]: same.

File: rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp [hunk 5]
```diff
@@ -430,11 +430,17 @@ error_code sys_ppu_thread_rename(u32 thread_id, vm::cptr<char> name)
 	}
 
 	constexpr u32 max_size = 27; // max size including null terminator
-	const auto pname = name.get_ptr();
+
+	std::string_view pname = name.get_ptr();
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp]: Imagine a case where threadname is found at the last 27 bytes of a page without null terminator, with unmapped memory after.
The previous solution would read 27 bytes and stopped correctly, this one access violates at determination the length of string.

Comment (Nekotekina) [reply rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp]: Fine fine, should be ok now

File: rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp [hunk 6]
```diff
@@ -430,11 +430,11 @@ error_code sys_ppu_thread_rename(u32 thread_id, vm::cptr<char> name)
...
 
 	constexpr u32 max_size = 27; // max size including null terminator
+
 	const auto pname = name.get_ptr();
 
-	// thread_ctrl name is not changed (TODO)
-	const std::string res = thread->ppu_name.assign(pname, std::find(pname, pname + max_size, '\0'));
-	sys_ppu_thread.warning(u8"sys_ppu_thread_rename(): Thread renamed to “%s”", res);
+	thread->ppu_tname.store(stx::shared_cptr<std::string>::make(pname, std::find(pname, pname + max_size, '\0')));
+	sys_ppu_thread.warning(u8"sys_ppu_thread_rename(): Thread renamed to “%s”", pname);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp]:
```suggestion
	sys_ppu_thread.warning(u8"sys_ppu_thread_rename(): Thread renamed to “%s”", *thread->ppu_tname.load());
```

Comment (Nekotekina) [reply rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp]: It was already changed, github is lagging

File: Utilities/Thread.cpp [hunk 7]
```diff
@@ -1774,13 +1776,15 @@ void thread_base::initialize(bool(*wait_cb)(const void*))
...
 #if defined(__APPLE__)
-	pthread_setname_np(m_name.get().substr(0, 15).c_str());
+	name.resize(std::min<std::size_t>(15, name.size()));
+	pthread_setname_np(name.c_str());
 #elif defined(__DragonFly__) || defined(__FreeBSD__) || defined(__OpenBSD__)
-	pthread_set_name_np(pthread_self(), m_name.get().c_str());
+	pthread_set_name_np(pthread_self(), name.c_str());
 #elif defined(__NetBSD__)
-	pthread_setname_np(pthread_self(), "%s", const_cast<char*>(m_name.get().c_str()));
+	pthread_setname_np(pthread_self(), "%s", const_cast<char*>(name.c_str()));
```
Comment (elad335) [inline Utilities/Thread.cpp]:
```suggestion
	pthread_setname_np(pthread_self(), "%s", name.data());
```
Because `name` is longer a const string.

**Comment by Nekotekina** [comment]
Tested and fixed MSVC compilation failure locally.

**Pull request merged**